python/lsst/pipe/tasks/subtractBrightStars.py

```diff
@@ -0,0 +1,359 @@
+
+        Parameters
+        ----------
+        calexp : `~lsst.afw.image.Exposure` or `~lsst.afw.image.MaskedImage`
+            Calibrated exposure.
+        skyCorr : `~lsst.afw.math.backgroundList.BackgroundList`
+            Full focal plane sky correction, obtained by running
+            `~lsst.pipe.drivers.skyCorrection.SkyCorrectionTask`.
+        """
+        if isinstance(calexp, Exposure):
+            calexp = calexp.getMaskedImage()
+        calexp -= skyCorr.getImage()
+
+    def scaleModel(self, model, star, inPlace=True, nb90Rots=0):
+        """Compute scaling factor to be applied to the extended PSF so that its
+        amplitude matches that of an individual star.
+
+        Parameters
+        ----------
+        model : `~lsst.afw.image.MaskedImageF`
+            The extended PSF model, shifted (and potentially warped) to match
+            the bright star's positioning.
+        star : `~lsst.meas.algorithms.brightStarStamps.BrightStarStamp`
+            A stamp centered on the bright star to be subtracted.
+        inPlace : `bool`
+            Whether the model should be scaled in place. Default is `True`.
+        nb90Rots : `int`
+            The number of 90-degrees rotations to apply to the star stamp.
+
+        Returns
+        -------
+        scalingFactor : `float`
+            The factor by which the model image should be multiplied for it
+            to be scaled to the input bright star.
+        """
+        if self.config.scalingType == "annularFlux":
+            scalingFactor = star.annularFlux
+        elif self.config.scalingType == "leastSquare":
+            if self.statsControl is None:
+                self._setUpStatistics(star.stamp_im.mask)
+            starIm = star.stamp_im.clone()
+            # Rotate the star postage stamp.
+            starIm = rotateImageBy90(starIm, nb90Rots)
+            # Reverse the prior star flux normalization ("unnormalize").
+            starIm *= star.annularFlux
+            # The estimator of the scalingFactor (f) that minimizes (Y-fX)^2
+            # is E[XY]/E[XX].
+            xy = starIm.clone()
+            xy.image.array *= model.image.array
+            xx = starIm.clone()
+            xx.image.array = model.image.array**2
+            # Compute the least squares scaling factor.
+            xySum = makeStatistics(xy, self.statsFlag, self.statsControl).getValue()
+            xxSum = makeStatistics(xx, self.statsFlag, self.statsControl).getValue()
+            scalingFactor = xySum / xxSum if xxSum else 0
+        else:
+            raise AttributeError(
+                f'Unknown scalingType "{self.config.scalingType}"; implemented options '
+                'are "annularFlux" and "leastSquare".'
+            )
+        if inPlace:
+            model.image *= scalingFactor
+        return scalingFactor
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        """Run a single quantum of the task."""
```

Comment:
You can remove this docstring and replace it with a comment "#Docstring inherited." If it didn't have a docstring, it'd inherit it automatically from the parent `PipelineTask` which is probably sufficient here.